src/lib.rs

```diff
@@ -1,59 +1,74 @@
-use serde_json::{Value};
+use serde_json::{Map, Value};
+use thiserror::Error;
 
-pub fn hydrate(base: Value, item: &mut Value) {
-    if item.is_null(){
-        *item = base.clone();
-    }
-    else if item.is_object() {
-        if item.as_object() == base.as_object(){
```

Comment:
I got rid of the equality check because it doesn't save any time, since the equality check forces an iteration through the object.

Reply:
I was thinking (hoping) the native checks to determine equality may be faster than iterating through chained calls to this function.

Reply:
Hm, it'd be an interesting question to explore. My statement 

> equality check forces an iteration through the object

isn't true now that I think about it, because equality-of-all-fields != iteration. Maybe? I'd have to benchmark to know for sure.